core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/Stream.scala

```diff
@@ -508,35 +385,20 @@ abstract class Stream[+T] { self =>
       val cancel = self.foreach(v => updateEventState(condition.apply(v)))
     }
   }
-
-  /**
-    * Adds a listener for elements of this stream. Callbacks will be executed
-    * whenever a new value is published in order of when the callbacks were added.
-    * Callbacks added first will be called first and callbacks added last will
-    * be called last.
-    *
-    * @param thunk the listener to be called on each published value
-    * @return a function to call to remove the listener
-    */
-  def foreach(thunk: T => Unit): Cancel = {
-    self.synchronized {
-      listeners = listeners :+ thunk
-    }
-
-    () => {
-      self.synchronized {
-        listeners = listeners.filterNot(_ eq thunk)
-      }
-    }
-  }
 }
 
 object Stream {
   def periodic[T](period: Time)(value: => T)(implicit clock: Clock): Stream[T] = {
     new Stream[T] { self =>
+      override def subscribeToParents(): Unit = {}
+      override def unsubscribeFromParents(): Unit = {}
+
       override val expectedPeriodicity = Periodic(period)
 
       override val originTimeStream = Some(new Stream[Time] {
```

Comment:
I think this needs to be persistent

Reply:
In this case, persistent doesn't make sense since there are no parents to subscribe or unsubscribe to.